File: backend/app/schemas/preference.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PreferenceCreate(BaseModel):
    candidate_id: uuid.UUID | None = None
    feedback_type: str
    hunter_comment: str
    weight_adjustment: dict | None = None


class PreferenceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    candidate_id: uuid.UUID | None
    feedback_type: str
    hunter_comment: str
    weight_adjustment: dict | None
    created_at: datetime

File: frontend/src/types/preference.ts
export interface PreferenceCreate {
  candidate_id?: string | null
  feedback_type: string
  hunter_comment: string
  weight_adjustment?: Record<string, number> | null
}

export interface PreferenceResponse {
  id: string
  project_id: string
  candidate_id: string | null
  feedback_type: string
  hunter_comment: string
  weight_adjustment: Record<string, number> | null
  created_at: string
}
